lib/ansible/modules/packaging/os/rhsm_repository.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (@giovannisciortino)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: rhsm_repository
+short_description: Manage RHSM repositories using the C(subscription-manager)
+                   command
+description:
+  - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription
+    Management entitlement platform using the C(subscription-manager) command.
+version_added: '2.5'
+author: Giovanni Sciortino (@giovannisciortino)
+notes:
+  - In order to manage RHSM repositories the system must be already registered
+    to RHSM manually or using the ansible module redhat_subscription.
+  - One option between name and list must be defined, both options in the
+    same task must not be defined.
+
+requirements:
+  - subscription-manager
+options:
+  state:
+    description:
+      - If state is equal to enabled or disabled, indicates the desired
+        repository state.
+      - If state is equal to list, list_enabled, or list_disabled, list
+        C(all)/C(enabled)/C(disabled) repositories.
+    choices: [enabled, disabled, list, list_enabled, list_disabled]
+    required: False
+    default: enabled
+  name:
+    description:
+      - The ID of repositories to enable.
+      - To operate on several repositories this can accept a comma separated
+        list or a YAML list.
+      - If state is equal to enabled or disabled, this option is required.
+      - If state is not equal to enabled or disabled, this option must not be used.
+    required: False
+    default: None
+'''
+
+EXAMPLES = '''
+- name: List all RHSM repositories.
+  rhsm_repository:
+    state: list
+  register: rhsm_repository
+
+- name: List enabled RHSM repositories.
+  rhsm_repository:
+    state: list_enabled
+  register: enabled_rhsm_repository
+
+- name: Enable a RHSM repository
+  rhsm_repository:
+    name: rhel-7-server-rpms
+
+- name: Disable all RHSM repositories
+  rhsm_repository:
+    name: '*'
+    state: disabled
+
+- name: Enable all repositories starting with rhel-6-server
+  rhsm_repository:
+    name: rhel-6-server*
+    state: enabled
+
+- name: Disable all repositories except rhel-7-server-rpms
+  rhsm_repository:
+    name: "{{ item }}"
+    state: disabled
+  with_items: "{{ rhsm_repository.repositories | map(attribute='id') | difference(['rhel-7-server-rpms']) }}"
```

Comment:
What about to format this like this:

```yaml
  with_items: "{{
    rhsm_repository.repositories |
    map(attribute='id') |
    difference(['rhel-7-server-rpms']) }}"
```

Reply:
Changed in the last commit, thanks for the review